// yx-translate/src/services/translate.service.js
import { Inject, Injectable } from '@angular/core';
import { Http } from "@angular/http";
var TranslateConfig = (function () {
    function TranslateConfig() {
    }
    return TranslateConfig;
}());
export { TranslateConfig };
var TranslateService = (function () {
    function TranslateService(translateConfig, http) {
        this.translateConfig = translateConfig;
        this.http = http;
        this.language = translateConfig.language;
        this.folder = translateConfig.folder;
        this.reloadTranslations();
    }
    TranslateService.prototype.reloadTranslations = function () {
        var _this = this;
        var savedLang = localStorage.getItem('yx_language');
        if (savedLang) {
            this.language = savedLang;
        }
        this.http.get(this.folder + '/' + this.language + '.json').subscribe(function (res) {
            _this.translations = res.json();
        });
    };
    TranslateService.prototype.translate = function (value) {
        if (!this.translations) {
            return value;
        }
        var translation = this.translations[value];
        if (translation) {
            return translation;
        }
        else {
            return value;
        }
    };
    TranslateService.prototype.changeLanguage = function (value) {
        if (value !== this.language) {
            localStorage.setItem('yx_language', value);
            location.reload();
        }
    };
    return TranslateService;
}());
export { TranslateService };
TranslateService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
TranslateService.ctorParameters = function () { return [
    { type: TranslateConfig, decorators: [{ type: Inject, args: ["translate.config",] },] },
    { type: Http, },
]; };
