// yx-translate/src/translate.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TranslatePipe } from './pipes/translate.pipe';
import { TranslateConfig, TranslateService } from "./services/translate.service";
var TranslateModule = (function () {
    function TranslateModule() {
    }
    TranslateModule.forRoot = function (language, folder) {
        var translateConfig = new TranslateConfig();
        translateConfig.folder = 'assets/yx-i18n';
        translateConfig.language = 'en';
        if (language) {
            translateConfig.language = language;
        }
        if (folder) {
            translateConfig.folder = folder;
        }
        return {
            ngModule: TranslateModule,
            providers: [
                TranslateService,
                { provide: "translate.config", useValue: translateConfig }
            ]
        };
    };
    return TranslateModule;
}());
export { TranslateModule };
TranslateModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                exports: [
                    TranslatePipe
                ],
                declarations: [TranslatePipe]
            },] },
];
/** @nocollapse */
TranslateModule.ctorParameters = function () { return []; };
